bigbluebutton/bigbluebutton
# feat(recording): adds custom userdata recording permission

### What does this PR do?

Adds two new userdata parameters:
 - `bbb_record_permission`: if true, bypass the moderator/viewer role permission check and allows the user to start/stop the recording. Also, if set to false will deny the user to record even if he is a moderator
 - `bbb_record_permission_tooltip`: a string to show in the tooltip when the user don't have permission, useful to explain why the user can't start recording

### Motivation
This enables integrations to have custom rules for specific users to have or not permission to record a meeting

### How to test
Set `userdata-bbb_record_permission` and `userdata-bbb_record_permission_tooltip` in custom parameters before joining and see the effect in meeting.

For example. setting
```
userdata-bbb_record_permission=false
userdata-bbb_record_permission_tooltip=You have used all your recording quota
```
and joining as moderator, the user will not be enabled to record and the tooltip will show the message:
![image](https://github.com/user-attachments/assets/39cbd85a-9655-41e1-ada0-79eb7dceddd8)
